edx_toggles/toggles/__init__.py

```diff
@@ -2,10 +2,9 @@
 Expose public feature toggle API.
 """
 from .internal.setting_toggle import SettingDictToggle, SettingToggle
-from .internal.waffle.legacy import WaffleFlag, WaffleFlagNamespace, WaffleSwitch, WaffleSwitchNamespace
-
-# Create waffle aliases for forward compatibility
-LegacyWaffleFlag = WaffleFlag
-LegacyWaffleFlagNamespace = WaffleFlagNamespace
-LegacyWaffleSwitch = WaffleSwitch
-LegacyWaffleSwitchNamespace = WaffleSwitchNamespace
+from .internal.waffle.flag import NonNamespacedWaffleFlag, WaffleFlag
+from .internal.waffle.legacy import WaffleFlag as LegacyWaffleFlag
```

Comment:
I noted in Slack that we might need an earlier change where `LegacyWaffleFlag` becomes a real class to get accurate custom attributes.  If that is the case, this will get updated.

Reply:
> FYI: I don’t think I can tell if we have stopped using the current WaffleFlag and are only using LegacyWaffleFlag (or futures), because both are coming from the legacy module with the class WaffleFlag when monitoring. (Apologies if I am missing something.)
> Proposal:
> 
>* Make LegacyWaffleFlag (and the other Legacy classes) a real class that subclasses the existing class, also in the legacy module.
> * Update the module based custom attributes to be statically defined custom attributes that let you know which is being imported. For example, we could set deprecated_current_waffle_class in the following legacy module classes: WaffleFlag, WaffleSwitch, WaffleFlagNamespace, and WaffleSwitchNamespace, still with a value including the class and toggle name. We could set deprecated_legacy_waffle_class in the new legacy module classes: LegacyWaffleFlag, LegacyWaffleSwitch, etc.
> * We would want to ensure that deprecated_current_waffle_class had no data before publishing edx-toggles 2.0.0. Note: If I don’t see any data, I might consider temporarily importing something from the wrong place, just to prove to myself that I am querying the data correctly.

If I understand correctly, we want to:

1. check which apps have statements such as `from edx_toggles.toggles import WaffleSwitch, WaffleFlag`.
2. replace these statements by `from edx_toggles.toggles import LegacyWaffleSwitch, LegacyWaffleFlag`

This is currently not possible because the class name indicated by `LegacyWaffleFlag.__class__` is "WaffleFlag".

Thus, a simple fix would be to write in edx-toggles==1.2.1 (to be released):

```
class LegacyWaffleFlag(WaffleFlag): pass
class LegacyWaffleSwitch(WaffleSwitch): pass
```

As far as I understand, it is unnecessary to modify the set_attribute calls. We would just need to check for custom "edx_toggles.toggles.waffle.legacy.WaffleFlag" and "edx_toggles.toggles.waffle.legacy.WaffleSwitch" custom attributes.

Reply:
Thanks, I think you are correct as long as the two sets of classes are defined in two different modules. As defined, we won’t see all issues, but just the last issue per transaction. It is ok if one issue masks another issue until the first is fixed, but not ok if a non-issue masks an issue and we ne we learn about it.

Reply:
Still awaiting response for separating these two sets of classes under different modules.

Reply:
Here is the monitoring PR: https://github.com/edx/edx-toggles/pull/98